xDrip/Scenes/Entries/EntriesList/EntriesListViewController.swift

```diff
@@ -59,9 +95,61 @@ class EntriesListViewController: UIViewController, EntriesListDisplayLogic {
         interactor?.doLoad(request: request)
     }
     
+    @IBAction private func onCancelButtonTap() {
+        let request = EntriesList.Cancel.Request()
+        interactor?.doCancel(request: request)
+    }
+    
+    @objc
+    private func onEditButtonTap() {
+        isEdit.toggle()
+        setupRightBarButtonItem()
+        tableView.isEditing = isEdit
+    }
+    
     // MARK: Display
     
     func displayLoad(viewModel: EntriesList.Load.ViewModel) {
+        sectionViewModels = viewModel.items
+        
+        setupTableView()
+    }
+    
+    private func setupUI() {
+        setupRightBarButtonItem()
+        
+        view.addBlur()
+    }
+    
+    private func setupRightBarButtonItem() {
+        let item = UIBarButtonItem(barButtonSystemItem: isEdit ? .done : .edit,
+                               target: self,
+                               action: #selector(onEditButtonTap))
+        
+        navigationItem.rightBarButtonItem = item
+    }
+    
+    private func setupTableView() {
+        tableController = EntriesListTableController(data: sectionViewModels)
```

Comment:
Why do you need to re-init it? Create it without a custom initializer, create a method to load new data.